Framework/DataHandling/src/ISISJournal.cpp

```diff
@@ -0,0 +1,322 @@
+  for (const auto &filename : filenames) {
+    auto cycle = convertFilenameToCycleName(filename);
+    if (!cycle.empty())
+      cycles.emplace_back(std::move(cycle));
+  }
+  return cycles;
+}
+} // namespace
+
+/** Construct the journal class for a specific instrument and cycle
+ * @param instrument : the ISIS instrument name to request data for e.g. "INTER"
+ * (case insensitive)
+ * @param cycle : the ISIS cycle the required data is from e.g. "19_4"
+ * @param internetHelper : class for sending internet requests
+ */
+ISISJournal::ISISJournal(std::string const &instrument,
+                         std::string const &cycle,
+                         std::unique_ptr<InternetHelper> internetHelper)
+    : m_internetHelper(std::move(internetHelper)),
+      m_runsFileURL(constructURL(instrument, cycle)),
+      m_indexFileURL(constructURL(instrument, INDEX_FILE_NAME)) {}
+
+ISISJournal::~ISISJournal() = default;
+
+ISISJournal::ISISJournal(ISISJournal &&rhs) = default;
+
+ISISJournal &ISISJournal::operator=(ISISJournal &&rhs) = default;
+
+/** Get the cycle names
+ *
+ * @returns : a list of all ISIS cycle names for the instrument
+ * @throws : if there was an error fetching the runs
+ */
+std::vector<std::string> ISISJournal::getCycleNames() {
+  if (!m_indexDocument) {
+    auto xmlString = getURLContents(m_indexFileURL);
+    m_indexDocument = parse(xmlString);
+  }
+  auto rootElement = m_indexDocument->documentElement();
+  throwIfNotValid(rootElement, JOURNAL_TAG);
+  auto filenames =
+      getAttributeForAllChildElements(rootElement, FILE_TAG, "name");
+  return convertFilenamesToCycleNames(filenames);
+}
+
+/** Get run names and other specified data for all runs that match the given
+ * filter criteria.
+ *
+ * @param valueNames : optional list of additional values to be returned e.g.
+ * "run_number", "title"
+ * @param filters : optional element names and values to filter the results by
+ * @throws : if there was an error fetching the runs
+ */
+std::vector<ISISJournal::RunData>
+ISISJournal::getRuns(std::vector<std::string> const &valueNames,
+                     ISISJournal::RunData const &filters) {
+  if (!m_runsDocument) {
+    auto xmlString = getURLContents(m_runsFileURL);
+    m_runsDocument = parse(xmlString);
+  }
+  auto rootElement = m_runsDocument->documentElement();
+  throwIfNotValid(rootElement, NXROOT_TAG);
+  return getValuesForAllElements(rootElement, valueNames, filters);
+}
+
+/** Get the contents of a file at a given URL
+ * @param url : the URL to fetch
+ * @returns : the contents of the file as a string
+ * @throws : if there was an error fetching the file
+ */
+std::string ISISJournal::getURLContents(std::string const &url) {
+  std::ostringstream serverReply;
+  auto const statusCode = m_internetHelper->sendRequest(url, serverReply);
+  if (statusCode != Poco::Net::HTTPResponse::HTTP_OK) {
+    throw Kernel::Exception::InternetError(
+        std::string("Failed to access journal file: ") +
```

Comment:
Could you add at the end `HTTP Code: ` before appending the status code so it's obvious the number is a status code